test/unit/bundle/lookup.js

```diff
@@ -55,4 +55,23 @@ describe("Bundle Lookup", function() {
     sr = new Cldr("sr-RS");
     expect(sr.attributes.bundle).to.equal("sr");
   });
+
+  it("should clear the global _availableBundleMapQueue on failure to process bundle", function() {
+    expect(() => {
+      Cldr.load({
+        main: { "xx-XX": {} }
+      });
+      expect(Cldr._availableBundleMapQueue).to.include.members(["xx-XX"]);
+      new Cldr("xx-XX");
+    }).to.throw();
+    expect(Cldr._availableBundleMapQueue).to.be.empty;
```

Comment:
I'm wondering if we could do this:

```js
expect(() => {
  Cldr.load(
    {
      main: { "xx-XX": {} }
    },
    {
      main: { sr: {} }
    }
  );
  expect(Cldr._availableBundleMapQueue).to.include.members(["xx-XX", "sr"]);
  new Cldr("xx-XX");
}).to.throw();
expect(Cldr._availableBundleMapQueue).to.include.members(["sr"]);
```

Reply:
Thanks for the review :)

On a higher level, what would you expect to happen? If we create an instance of sr instead of xx-XX it would still throw because it would iterate in order. The test as it is would succeed and a second attempt of creating an instance of sr would succeed with the data parsed. Does this sound like an acceptable behaviour to you?

Reply:
> On a higher level, what would you expect to happen? If we create an instance of sr instead of xx-XX it would still throw because it would iterate in order

That's a good question. Yeap, I'd expect that to still throw. That being said, ideally we have a better error message saying it's missing supplemental.likelySubtags data for xx-XX.

Reply:
> The test as it is would succeed and a second attempt of creating an instance of sr would succeed with the data parsed. Does this sound like an acceptable behaviour to you?

Yes, though I am still not convinced about the solution. Do we really need this code change, or all we need is a better error message (comment above) and simply expect Cldrjs user to feed it on the right data?

The root cause of this issue is the artificial `{main: {"xx-XX": {...}}` without a proper likelySubtags. Similar issue could happen for CLDR data if one loads main without loading likelySubtags. Ideally, the root cause is fixed, i.e., user (i.e., engineer/developer) should pass the right data to CLDR.load.

What do you think?

Reply:
I agree with the root cause, however developers make mistakes and bad data would be passed sooner or later. I would expect to still be able to use the library for other data that is already loaded and newly loaded valid data rather than getting an error on anything after the invalid one is loaded. Currently the whole service needs to be recycled before it can serve requests again.

I am all for adding a more descriptive error message. I can manually throw a descriptive message between line 24 and 25 if coreLikelySubtags returns undefined which would help future debugging.

I guess we can have two approaches here -
1. load all data that can be loaded, collect all errors and throw at the end after everything has been processed
2. load everything that can be loaded until an error is observed. In this case stop iterating and remove everything from the queue including the failed bundle

Reply:
So let's try-catch that error, add a more descriptive error message and splice `Cldr._availableBundleMapQueue` to remove the failing item. What do you think?

Reply:
Yeah, happy with this approach. And just to confirm this means it will stop processing bundles on error, but next time it will succeed as the erroneous bundle will be removed and it will start from the one after?

Reply:
Exactly this. Thanks